addons/spreadsheet/static/src/global_filters/helpers.js

```diff
@@ -81,46 +70,43 @@ export function getValidGranularities(dateFilterValue) {
     }
 }
 
+export function getDateGlobalFilterRegistryItem(value) {
+    const registryKey = value.type === "relative" ? value.period : value.type;
+    return globalFilterDateRegistry.get(registryKey);
+}
+
+export function getDateGlobalFilterValueFromDefault(value) {
+    if (!value || typeof value !== "string") {
+        return undefined;
+    }
+    const now = DateTime.local();
+    const registryItem = globalFilterDateRegistry.get(value);
+    return registryItem.canOnlyBeDefault
+        ? registryItem.getDefaultValue(now)
+        : { type: "relative", period: value };
+}
+
+export function getDateGlobalFilterTypes() {
+    return globalFilterDateRegistry
+        .getKeys()
+        .sort(
+            (a, b) =>
+                globalFilterDateRegistry.get(a).sequence - globalFilterDateRegistry.get(b).sequence
+        );
+}
+
 /**
  * Compute the display name of a date filter value.
  */
 export function dateFilterValueToString(value) {
-    switch (value?.type) {
-        case "relative":
-            return RELATIVE_PERIODS[value.period];
-        case "month": {
-            const month = DateTime.local().set({
-                year: value.year,
-                month: value.month,
-            });
-            return month.toFormat("LLLL yyyy");
-        }
-        case "quarter": {
-            return _t("Q%(quarter)s %(year)s", {
-                quarter: value.quarter,
-                year: value.year,
-            });
-        }
-        case "year":
-            return String(value.year);
-        case "range":
-            if (value.from && value.to) {
-                const interval = Interval.fromDateTimes(
-                    DateTime.fromISO(value.from).startOf("day"),
-                    DateTime.fromISO(value.to).endOf("day")
-                );
-                return interval.toLocaleString(DateTime.DATE_FULL);
-            } else if (value.from) {
-                return _t("Since %(from)s", {
-                    from: DateTime.fromISO(value.from).toLocaleString(DateTime.DATE_FULL),
-                });
-            } else if (value.to) {
-                return _t("Until %(to)s", {
-                    to: DateTime.fromISO(value.to).toLocaleString(DateTime.DATE_FULL),
-                });
-            }
+    if (!value) {
+        return _t("All time");
+    }
+    try {
+        return getDateGlobalFilterRegistryItem(value).getValueString(value);
+    } catch {
+        return _t("All time");
```

Comment:
what kind of exception can we have here ?
Isn't this hiding crap under the carpet ?